src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -315,10 +371,23 @@ private void check() {
                 return;
             }
             if (workspace == null) {
+                recurrencePeriod = Math.min((long) (recurrencePeriod * RECURRENCE_PERIOD_BACKOFF), MAX_RECURRENCE_PERIOD);
                 return; // slave not yet ready, wait for another day
             }
             TaskListener listener = listener();
             try (Timeout timeout = Timeout.limit(10, TimeUnit.SECONDS)) {
+                if (watching) {
+                    Integer exitCode = controller.exitStatus(workspace, launcher());
+                    if (exitCode == null) {
+                        LOGGER.log(Level.FINE, "still running in {0} on {1}", new Object[] {remote, node});
+                    } else if (awaitingAsynchExit) {
+                        recurrencePeriod = 0;
```

Comment:
Maybe safer to have a *slight* nonzero delay?

Reply:
No, (exact) zero is the signal to avoid rescheduling.

Reply:
Ah, so it's a special signal value, normally that would be "-1".

Reply:
Perhaps. At any rate, zero has been used for this purpose from the start—not introduced in this patch. I can make that explicit in Javadoc.